taskotron_python_versions/python_usage.py

```diff
@@ -1,55 +1,59 @@
-from .common import log, write_to_artifact, file_contains
+from .common import log, write_to_artifact
 
 
-WARNING = 'DEPRECATION WARNING: python2 invoked with /usr/bin/python'
+MESSAGE = """The following packages (Build)Require `/usr/bin/python`
+(or `python-unversioned-command`):
 
-MESSAGE = """You've used /usr/bin/python during build on the following arches:
-
-  {{}}
+  * {}
 
 Use /usr/bin/python3 or /usr/bin/python2 explicitly.
 /usr/bin/python will be removed or switched to Python 3 in the future.
-
-Grep the build.log for the following to find out where:
-
-    {}
-""".format(WARNING)
+"""
 
 INFO_URL = ('https://fedoraproject.org/wiki/Changes/'
-            'Avoid_usr_bin_python_in_RPM_Build')
+            'Move_usr_bin_python_into_separate_package')
 
+PYTHON_COMMAND = (
+    '/usr/bin/python',
+    'python-unversioned-command',
+)
 
-def task_python_usage(logs, koji_build, artifact):
-    """Parses the build.logs for /usr/bin/python invocation warning
+
+def task_python_usage(packages, koji_build, artifact):
+    """Check if the packages depend on /usr/bin/python.
     """
     # libtaskotron is not available on Python 3, so we do it inside
     # to make the above functions testable anyway
     from libtaskotron import check
 
     outcome = 'PASSED'
 
-    problem_arches = set()
+    problem_rpms = set()
+
+    for package in packages:
+        log.debug('Checking {}'.format(package.filename))
 
-    for buildlog in logs:  # not "log" because we use that name for logging
-        log.debug('Will parse {}'.format(buildlog))
+        for name in package.require_names:
+            name = name.decode()
 
-        if file_contains(buildlog, WARNING):
-            log.debug('{} contains our warning'.format(buildlog))
-            _, _, arch = buildlog.rpartition('.')
-            problem_arches.add(arch)
-            outcome = 'FAILED'
+            if name in PYTHON_COMMAND:
+                log.error(
+                    '{} requires {}'.format(package.filename, name))
+                problem_rpms.add(package.nvr)
```

Comment:
do we have nevra here? that would tell the minatainer if it is src or built package that requires it I guess.

Reply:
I have used the filename here, it shows if the package is src or built.